src/components/Dropdown.tsx

```diff
@@ -0,0 +1,288 @@
+
+  // Destructure common state
+  const {
+    isDropdownOpen,
+  } = state;
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+      dispatch({ type: ActionType.ToggleDropdown });
+    }
+  };
+
+  /*------------------------------------------------------------------------*/
+  /* ------------------------- Lifecycle Functions ------------------------ */
+  /*------------------------------------------------------------------------*/
+
+  /**
+   * Mount
+   * @author Yuen Ler Chow
+   */
+  useEffect(() => {
+    // Add event listener to close dropdown when clicking outside
+    if (isDropdownOpen) {
+      document.addEventListener('mousedown', handleClickOutside);
+    } else {
+      // Remove event listener when dropdown is closed
+      document.removeEventListener('mousedown', handleClickOutside);
+    }
+
+    // Cleanup
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [isDropdownOpen]);
+
+  /*----------------------------------------*/
+  /* --------------- Main UI -------------- */
+  /*----------------------------------------*/
+
+  return (
+    <div className="dropdown" ref={dropdownRef}>
+      <button
+        className={
+          combineClassNames([
+            'btn dropdown-toggle border',
+            isDropdownOpen && 'show',
+            variant && variant === Variant.Dark && 'btn-dark text-light',
+            variant && variant === Variant.Light && 'btn-light text-dark',
+          ])
+        }
+        type="button"
+        id={dropdownButton.id}
+        aria-expanded={isDropdownOpen}
+        aria-label={dropdownButton.ariaLabel}
+        onClick={() => {
+          dispatch({
+            type: ActionType.ToggleDropdown,
+          });
+        }}
+      >
+        {
+          dropdownButton.content
+        }
+      </button>
+      <ul className={
+          combineClassNames([
+            'dropdown-menu',
+            variant && variant === Variant.Dark && 'dropdown-menu-dark',
+            isDropdownOpen && 'show',
+          ])
+        }
+      >
+        {Object.values(items).map((item) => {
+          if (item.type === DropdownItemType.Header) {
+            return (
+              // TODO: Implement header
+              <span />
+            );
+          }
+          if (item.type === DropdownItemType.Divider) {
+            return (
+              // TODO: Implement divider
+              <span />
+            );
+          }
+          return (
+            <li
+              key={item.id}
+            >
+              <button
+                type="button"
+                aria-label={item.ariaLabel}
+                className="dropdown-item"
+                onClick={(e) => {
+                  e.preventDefault();
+                  dispatch({
+                    type: ActionType.ToggleDropdown,
```

Comment:
To be very clear here, can you switch this over to the `CloseDropdown` action as well?